features/FEATURE_BLE/ble/BLETypes.h

```diff
@@ -32,6 +33,237 @@
 
 namespace ble {
 
+/* replace with std::clamp when it arrives */
+template<typename T, typename R>
+void clamp(T& value, const R& min, const R& max) {
+    if (value > max) {
+        value = max;
+    } else if (value < min) {
+        value = min;
+    }
+}
+
+template<uint32_t Min, uint32_t Max>
+struct Range {
+    enum {
+        MIN = Min,
+        MAX = Max
+    };
+};
+
+template<typename Rep>
+struct DefaultRange;
+
+template<>
+struct DefaultRange<uint8_t> {
+    typedef Range<0, 0xFF> type;
+};
+
+template<>
+struct DefaultRange<uint16_t > {
+    typedef Range<0, 0xFFFF> type;
+};
+
+template<>
+struct DefaultRange<uint32_t> {
+    typedef Range<0, 0xFFFFFFFF> type;
+};
+
+
+template<typename Rep, uint32_t TB, typename Range = typename DefaultRange<Rep>::type >
+struct Duration {
```

Comment:
put this in its own header, this isn't ble specific and would be useful to other people too!

Reply:
Done.

Would be nice to have it in platform some day but it requires move refinement.